src/twisted/internet/test/reactormixins.py

```diff
@@ -324,26 +326,30 @@ def stop():
             timedOutCall.cancel()
 
 
-    def makeTestCaseClasses(cls):
+    @classmethod
+    def makeTestCaseClasses(
+        cls: Type['ReactorBuilder']
+    ) -> Dict[str, Union[Type['ReactorBuilder'], Type['SynchronousTestCase']]]:
         """
         Create a L{SynchronousTestCase} subclass which mixes in C{cls} for each
         known reactor and return a dict mapping their names to them.
         """
-        classes = {}
+        classes = {}  # type: Dict[str, Union[Type['ReactorBuilder'], Type['SynchronousTestCase']]]   # noqa
```

Comment:
Quoting in type *comments* seems doubly pointless, even if it works?

Reply:
I'm not sure exactly how mypy parses the file and comments to do all the type checks.  I just kept the use the same to be consistent.

Reply:
For mypy's sake you never need to quote, but the Python VM does require it when using a type name before it is defined, unless you use `from __future__ import annotations`, but that's a Python 3.7 feature.

Reply:
@mthuurne Yes, for now I need to quote `'ReactorBuilder'`.  Since Twisted still supports Python 3.5 and 3.6, I cannot use `from __future__ import annotations`